skfs/src/Context.sk

```diff
@@ -1691,6 +1702,14 @@ fun withRegionValue<T>(f: () ~> T): T {
   }
 }
 
+@debug
+fun callOnParentObstack<T>(saved: Obstack, f: () -> T): T {
+  switchToParent(saved);
+  res = f();
```

Comment:
I guess we get in to all sorts of trouble here if f creates or destroys an obstack? Probably not an easy way of guarding against this. But maybe a big warning comment couldn't hurt?